tests/dragonfly/replication_test.py

```diff
@@ -1044,3 +1044,63 @@ async def test_readonly_script(df_local_factory):
         assert False
     except aioredis.ResponseError as roe:
         assert 'READONLY ' in str(roe)
+
+
+cases = [
+    [2, 2],
+    [2, 4],
+    [4, 2],
+    [8, 8],
+]
+
+
+@dfly_args({"proactor_threads": 2})
+@pytest.mark.parametrize("master_threads, replica_threads", cases)
+@pytest.mark.asyncio
+async def test_take_over(df_local_factory, master_threads, replica_threads):
+    master = df_local_factory.create(proactor_threads=master_threads,
+                                     port=BASE_PORT,
+                                     #  vmodule="journal_slice=2,dflycmd=2,main_service=1",
+                                     logtostderr=True)
+    replica = df_local_factory.create(
+        port=BASE_PORT+1, proactor_threads=replica_threads)
+    df_local_factory.start_all([master, replica])
+    c_master = aioredis.Redis(port=master.port)
+    c_replica = aioredis.Redis(port=replica.port)
+    c_blocking = aioredis.Redis(port=master.port)
+
+    await c_replica.execute_command(f"REPLICAOF localhost {master.port}")
+    await wait_available_async(c_replica)
+
+    async def counter(key):
+        value = 0
+        await c_master.execute_command(f"SET {key} 0")
+        while True:
+            try:
+                value = await c_master.execute_command(f"INCR {key}")
+            except (redis.exceptions.ConnectionError, redis.exceptions.ResponseError) as e:
+                break
```

Comment:
I think we want to make sure in the test that we got inside this exception

Reply:
done